electron/extraction.js

```diff
@@ -0,0 +1,40 @@
+// The code below is a copy of cli.js from mcode-extraction-framework
+
+const { MCODEClient } = require('../node_modules/mcode-extraction-framework/src/client/MCODEClient');
+const logger = require('../node_modules/mcode-extraction-framework/src/helpers/logger');
+const { mcodeApp } = require('../node_modules/mcode-extraction-framework/src/cli/app');
+
+async function runExtraction(fromDate, toDate, configFilepath, runLogFilepath, debug, allEntries) {
+  try {
+    let defaultFromDate = fromDate;
+    let defaultToDate = toDate;
+    let defaultDebug = debug;
+    if (fromDate.length < 1) {
+      defaultFromDate = undefined;
+    }
+    if (toDate.length < 1) {
+      defaultToDate = undefined;
+    }
+    if (!debug) {
+      defaultDebug = undefined;
+    }
+    const extractedData = await mcodeApp(
+      MCODEClient,
+      defaultFromDate,
+      defaultToDate,
+      configFilepath,
+      runLogFilepath,
+      defaultDebug,
+      allEntries,
+    ).then((value) => value);
+    return extractedData;
+  } catch (e) {
+    if (debug) logger.level = 'debug';
+    logger.error(e.message);
+    logger.debug(e.stack);
+    process.exit(1);
```

Comment:
I just thought of one more thing. I'm not sure if we want to have the `process.exit(1)` here. If we do happen to have an error during extraction, the process.exit here will close the electron app completely. If we do hit an error, we probably want a way for the user to start the extraction process over again without needing to open the app again.

Reply:
We could have it redirect to an error page that tells the user what the problem was if that information is available, and gives them an option to go back to the extraction form. I could add that once the output display is done.

Reply:
Yeah I like that idea! I think it's fine to either include that page as part of the output task or as a separate task entirely. For this PR, I think removing this line and using the "Reset" button you already have should be sufficient!

Reply:
I think that adding code for another button / display would take almost as much work as implementing the error page itself, but I couldn't implement the error page in this version because Extract is still a class component. Maybe I could just change the line to modify the value of this.state.submitted? That will achieve the same thing as hitting the "reset" button.

Reply:
Oh I had just meant you could remove this line here and then the electron app wouldn't close. I think that would leave the app in the "running extraction..." state for now. And a user could use your existing "Reset" button to navigate back for now. Then we can implement the error page later on. Does that make sense?